pkg/kubelet/status/state/state_checkpoint.go

```diff
@@ -177,3 +177,51 @@ func (sc *stateCheckpoint) ClearState() error {
 	sc.cache.ClearState()
 	return sc.storeState()
 }
+
+type noopStateCheckpoint struct{}
+
+// Dummy state checkpoint manager
```

Comment:
nit: start with the name of the func

```suggestion
// NewNoopStateCheckpoint creates a dummy state checkpoint manager
```